src/pyscipopt/scip.pyx

```diff
@@ -1138,13 +1170,30 @@ cdef class Model:
         cons -- the linear constraint
         """
         # TODO this should ideally be handled on the SCIP side
-        dual = None
+        cdef int _nvars
+        cdef SCIP_VAR** _vars
+        cdef SCIP_Bool _success
+        dual = 0.0
         try:
+            PY_SCIP_CALL(SCIPgetConsNVars(self._scip, cons.cons, &_nvars, &_success))
             if cons.isOriginal():
                 transcons = <Constraint>self.getTransformedCons(cons)
+            else:
+                transcons = cons
+            if _nvars > 1:
                 dual = SCIPgetDualsolLinear(self._scip, transcons.cons)
             else:
-                dual = SCIPgetDualsolLinear(self._scip, cons.cons)
+                _vars = <SCIP_VAR**> malloc(sizeof(SCIP_VAR*))
+                PY_SCIP_CALL(SCIPgetConsVars(self._scip, transcons.cons, _vars, _nvars, &_success))
+                LPsol = SCIPvarGetLPSol(_vars[0])
+                rhs = SCIPgetRhsLinear(self._scip, transcons.cons)
+                lhs = SCIPgetLhsLinear(self._scip, transcons.cons)
+                if (LPsol == rhs) or (LPsol == lhs):
+                    dual = SCIPgetVarRedcost(self._scip, _vars[0])
+                    print(LPsol, rhs, lhs)
+                else:
+                    dual = 0
+
```

Comment:
mmm There is a function `SCIPgetVarsLinear` that gives you the vars right away (so you don't need the malloc)

Also, should `SCIPgetVarRedcost` be 0 when `LPsol != rhs` and `LPsol != lhs` ?

The print should be removed